packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -88,5 +95,64 @@ test.describe('`Pie Modal is closed`', () => {
             expect(messages).toHaveLength(1);
         });
     });
-});
 
+    test('should return focus to specified element', async ({ mount, page }) => {
+        // Arrange
+        const component = renderTestPieModal({
+            returnFocusAfterCloseSelector: '#focus-me',
+        });
+
+        await mount(WebComponentTestWrapper, {
+            props: {
+                pageMode: true,
+            },
+            slots: {
+                component,
+                pageMarkup: `<div>
+                    <button id="default"></button>
+                    <button id="focus-me"></button>
+                    <button id="not-me"></button>
+                </div>`,
+            },
+        });
+
+        // Act
+        await page.locator(closeButtonSelector).click();
+
+        const focusedElement = await page.locator(':focus');
+        const focusedElementId = await focusedElement.getAttribute('id');
+
+        // Assert
+        expect(focusedElementId).toBe('focus-me');
+    });
+
+    test('should return focus to first matching element', async ({ page, mount }) => {
```

Comment:
Do you need a test for when the modal defaults it focus to the opening element?

Reply:
I did try this yesterday and couldn't get it working, but I've just made some changes around the focus so I'll give it another go and maybe it'll work this time